apps/prairielearn/src/middlewares/authzAuthnHasCoursePreviewOrInstanceView.ts

```diff
@@ -1,6 +1,17 @@
-import { createAuthzMiddleware } from './authzHelper.js';
+import { type NextFunction, type Request, type Response } from 'express';
 
-export default createAuthzMiddleware({
-  oneOfPermissions: ['has_course_permission_preview', 'has_course_instance_permission_view'],
-  unauthorizedUsers: 'block',
-});
+import { HttpStatusError } from '@prairielearn/error';
+
+export default function (req: Request, res: Response, next: NextFunction) {
```

Comment:
This also affects all `/pl/course_instance/:course_instance_id(\\d+)/instructor` routes. Is that intentional?

Reply:
In other words, the effective user check continues to use the same middleware. The difference now is that a check (that already existed before #12496) for authenticated user is restored, and (rightfully) provides a proper 403 since there is no reason to show the helper page, since the current user is not actually a staff member.

Reply:
I see 👍